drivers/pwm/pwm_mcux.c

```diff
@@ -135,23 +141,42 @@ static int pwm_mcux_init(const struct device *dev)
 	struct pwm_mcux_data *data = dev->data;
 	pwm_config_t pwm_config;
 	status_t status;
-	int i, err;
+	int i;
 
 	if (!device_is_ready(config->clock_dev)) {
 		LOG_ERR("clock control device not ready");
 		return -ENODEV;
 	}
 
-	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+#ifdef CONFIG_PINCTRL
+	int err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
 	if (err < 0) {
 		return err;
 	}
-
+#endif
 	PWM_GetDefaultConfig(&pwm_config);
 	pwm_config.prescale = config->prescale;
 	pwm_config.reloadLogic = kPWM_ReloadPwmFullCycle;
 	pwm_config.clockSource = kPWM_BusClock;
 
+	/* Set the Submodule Clocks for FlexPWM */
+#if CONFIG_BOARD_LPCXPRESSO55S36
```

Comment:
We shouldn't have board specific code in a driver

Reply:
Probably better placed in `soc/arm/nxp_lpc/lpc55xxx/soc.c` in `clock_init`

Reply:
Ahh I see, thank you for the suggestion, will be fixed right away!